src/transformers/models/encodec/modeling_encodec.py

```diff
@@ -762,6 +803,24 @@ def decode(
             return (audio_values,)
         return EncodecDecoderOutput(audio_values)
 
+    def compute_mel_spectrogram(self, audio, n_fft, hop_length, n_mels=64):
+        device = audio.device
+        # Adjust n_mels if necessary to avoid warnings
+        n_mels = min(n_mels, n_fft // 2 + 1)
+        # Create the window function on the correct device
+        window = torch.hann_window(n_fft, device=device)
```

Comment:
(nit)
```suggestion
        # Adjust n_mels if necessary to avoid warnings
        n_mels = min(n_mels, n_fft // 2 + 1)
        # Create the window function on the correct device
        window = torch.hann_window(n_fft, device=audio.device)
```